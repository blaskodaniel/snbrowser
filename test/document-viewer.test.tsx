import { shallow } from 'enzyme'
import React from 'react'
import { act } from 'react-dom/test-utils'
import { Repository } from '@sensenet/client-core'
import { RepositoryContext } from '@sensenet/hooks-react'
import DocviewerComponent from '../src/components/document-viewer'
import { TestContentCollection } from './_mocks_/test_contents'

describe('Document viewer instance', () => {
  let wrapper: any
  let repo: any

  beforeEach(() => {
    window.fetch = function fetchMethod() {
      return Promise.resolve({ d: TestContentCollection } as any)
    }
    repo = new Repository()
    repo.loadCollection = function fetchMethod() {
      return Promise.resolve({ d: { results: TestContentCollection } } as any)
    }
  })

  it('should renders correctly', async () => {
    const props = {
      documentId: TestContentCollection[0].Id.toString(),
    }
    await act(async () => {
      wrapper = shallow(
        <RepositoryContext.Provider value={repo}>
          <DocviewerComponent {...props} />
        </RepositoryContext.Provider>,
      )
    })
    console.log(wrapper.update().debug())

    expect(wrapper.update()).toMatchSnapshot()
  })
})
