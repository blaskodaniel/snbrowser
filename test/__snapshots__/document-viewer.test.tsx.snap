// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document viewer instance should renders correctly 1`] = `
<withRouter(DocViewer)
  documentId="5908"
/>
`;
